pkg/config/store/yaml/store.go

```diff
@@ -90,7 +90,7 @@ func (s *yamlStore) LoadFrom(path string) error {
 	var cfg config.Config
 	cfg, err = config.New(versioned.Version)
 	if err != nil {
-		return store.LoadError{Err: fmt.Errorf("unable to create config for version %q: %w", versioned.Version, err)}
+		return store.LoadError{Err: config.UnsupportedVersionError{Version: versioned.Version}}
```

Comment:
`config.New()` should be in charge of returning `UnsupportedVersionError` since this caller does not know why the error was emitted.

Reply:
@estroz `config.New()` does return the UnsupportedVersionError already: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/config/registry.go#L34

Reply:
So `err` in the original code: `fmt.Errorf("unable to create config for version %q: %w", versioned.Version, err)` is an UnsupportedVersionError . So I can totally just return the err as is.